internal/clusterfeature/features/vault/manager.go

```diff
@@ -127,6 +130,13 @@ func (m FeatureManager) ValidateSpec(ctx context.Context, spec clusterfeature.Fe
 		return err
 	}
 
+	if !m.isManagedEnabled && !vaultSpec.CustomVault.Enabled {
+		return clusterfeature.InvalidFeatureSpecError{
+			FeatureName: featureName,
+			Problem:     "Pipeline's Vault configuration is not supported",
```

Comment:
I think this is not entirely correct, a little complicated and not very helpful. I'd rephrase it to something like this:
```suggestion
			Problem:     "Pipeline's managed Vault service is not available, configure a custom Vault instance",
```